### PR TITLE
User Authentication | Logging out the user

### PR DESCRIPTION
**Pull request by fiyinfoluwaafol**
At the backend, created the route handler for user logout, which essentially destroys the session the user is logged into by clearing the cookies associated with that session.

At the frontend, created a button to trigger the logout POST request implemented in the backend server.

The video shows a test plan of the implemented logout feature
https://github.com/ShareSync/share-cal/assets/112602670/82979f2c-6123-4207-8223-4fdc977e9579

### CODE REVIEW

File: frontend/src/components/CalendarView/CalendarView.jsx [hunk 1]
```diff
@@ -1,10 +1,17 @@
 import "./CalendarView.css"
+import Header from "../Header/Header"
+import { UserContext } from '../../UserContext.js';
+import { useState, useContext } from "react";
 
 function CalendarView () {
+    const userInfo = useContext(UserContext).user;
+    console.log({userInfo});
```
Comment (juancafe4) [inline frontend/src/components/CalendarView/CalendarView.jsx:8]: remove this

File: frontend/src/components/Header/Header.jsx [hunk 2]
```diff
@@ -1,10 +1,31 @@
...
+
+    const handleOnLogout = async (event) => {
+        try {
+            await fetch(`http://localhost:3000/logout`, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                credentials: 'include'
+            });
```
Comment (juancafe4) [inline frontend/src/components/Header/Header.jsx:16]: move this call to a another file with util functions

File: frontend/src/components/Header/Header.css [hunk 3]
```diff
@@ -1,5 +1,6 @@
 #header {
     display: flex;
+    justify-content: center;
```
Comment (jameskim08) [inline frontend/src/components/Header/Header.css:3]: delete the comments below this line

File: frontend/src/components/Header/Header.jsx [hunk 4]
```diff
@@ -1,10 +1,31 @@
...
+
+            updateUser(null);
+        } catch (error) {
+            console.error('Error logging out:', error.message);
```
Comment (jameskim08) [inline frontend/src/components/Header/Header.jsx:20]: you should provide an error message to the user here as well.

**Pull request merged**